reliquery/storage.py

```diff
@@ -505,6 +511,463 @@ def get_all_relic_data(self) -> List[Dict]:
+            # Create the folder with the previous metadata
+            folder = (
+                self.service.files().create(body=folder_metadata, fields="id").execute()
+            )
+            # Get folder id for the folder just created
+            folder_id = "{}".format(folder.get("id"))
+
+            # Retrieve existing parents of the folder
+            folder = (
+                self.service.files()
+                .get(fileId="{}".format(folder_id), fields="parents")
+                .execute()
+            )
+            previous_parents = ",".join(folder.get("parents"))
+
+            # Remove the previous parents and replace with the parent we want
+            folder = (
+                self.service.files()
+                .update(
+                    fileId=folder_id,
+                    addParents=self.shared_folder_id,
+                    removeParents=previous_parents,
+                    fields="id, parents",
+                )
+                .execute()
+            )
+
+            folder = (
+                self.service.files().create(body=folder_metadata, fields="id").execute()
+            )
+            self.root_id = folder.get("id")
+
+    def _join_path(self, path: StoragePath) -> str:
+        return "/".join([self.prefix] + path)
+
+    def _create_folder(self, path_name, parent_id):
+        # Create folder metadata
+        folder1_metadata = {
+            "name": path_name,
+            "mimeType": "application/vnd.google-apps.folder",
+            "parents": "{}".format([parent_id]),
+        }
+        # Create the folder with the previous metadata
+        folder1 = (
+            self.service.files().create(body=folder1_metadata, fields="id").execute()
+        )
+        # Get folder id for the folder just created
+        folder1_id = "{}".format(folder1.get("id"))
+
+        # Retrieve existing parents of the folder
+        folder = (
+            self.service.files()
+            .get(fileId="{}".format(folder1_id), fields="parents")
+            .execute()
+        )
+        previous_parents = ",".join(folder.get("parents"))
+
+        # Remove the previous parents and replace with the parent we want
+        folder = (
+            self.service.files()
+            .update(
+                fileId=folder1_id,
+                addParents=parent_id,
+                removeParents=previous_parents,
+                fields="id, parents",
+            )
+            .execute()
+        )
+
+        return folder
+
+    def _list_items_in_folder(self, folder_id):
+        # List files in parent folder
+        results = (
+            self.service.files()
+            .list(
+                q="parents in '{}'".format(folder_id),
+                pageSize=100,
+                fields="nextPageToken, files(id, name)",
+            )
+            .execute()
+        )
+        items = results.get("files", [])
+        return items
+
+    def _find_id_in_folder(self, folder_id, fileName):
```

Comment:
can there be files that share the same filename with differing ids? If so this will only return the first one it finds not all by the same file name no? If not then ignore this.